x-pack/docs/en/security/securing-communications/security-basic-setup-https.asciidoc

```diff
@@ -0,0 +1,595 @@
+
+   b. When asked if you want to use an existing CA, enter `y`.
+
+   c. Enter the path to your CA. This is the absolute path to
+   the `elastic-stack-ca.p12` file that you generated for your cluster.
+
+   d. Enter the password for your CA.
+
+   e. Enter an expiration value for your certificate. You can enter the
+   validity period in years, months, or days. For example, enter `90D` for 90
+   days.
+
+   f. When asked if you want to generate one certificate per node, enter `y`.
++
+Each certificate will have its own private key, and will be issued for a
+specific hostname or IP address.
+
+   g. When prompted, enter the name of the first node in your cluster. Use the same node name from previous steps.
+
+   h. Enter all hostnames used to connect to your first node. These hostnames
+   will be added as DNS names in the Subject Alternative Name (SAN) field in your certificate.
++
+List every hostname and variant used to connect to your cluster over HTTPS.
+
+   i. Enter the IP addresses that clients can use to connect to your node.
+
+   j. Repeat these steps for each additional node in your cluster.
+
+. After generating a certificate for each of your nodes, enter a password for
+   your private key when prompted.
+
+. Unzip the generated `elasticsearch-ssl-http.zip` file. This compressed file
+   contains one directory for both {es} and {kib}.
++
+--
+[source,txt]
+----
+/elasticsearch
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+
+[source,txt]
+----
+/kibana
+|_ README.txt
+|_ elasticsearch-ca.pem
+|_ sample-kibana.yml
+----
+--
+
+. Copy the relevant `http.p12` certificate to the `ES_PATH_CONF` directory on each node.
+
+. On each node, edit the `elasticsearch.yml` file to enable HTTPS security and
+   specify the location of the `http.p12` security certificate.
++
+[source,yaml]
+----
+xpack.security.http.ssl.enabled: true
+xpack.security.http.ssl.keystore.path: http.p12
+----
+
+. Add the password for your private key to a secure setting in {es}.
++
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.http.ssl.keystore.secure_password
+----
+
+. Restart {es}.
+
+**Next**: <<encrypt-kibana-http,Encrypt HTTP client communications for {kib}>>
+
+[[encrypt-kibana-http]]
+==== Encrypt HTTP client communications for {kib}
+
+Browsers send traffic to {kib} and {kib} sends traffic to {es}.
+These communication channels are configured separately to use TLS. You encrypt
+traffic between your browser and {kib}, and then encrypt traffic between
+{kib} and {es}.
+
+[[encrypt-kibana-browser]]
+===== Encrypt traffic between your browser and {kib}
+
+You create a server certificate and private key for {kib}. {kib} uses this
+server certificate and corresponding private key when receiving connections
+from web browsers.
+
+When you obtain a server certificate, you must set its subject alternative
+name (SAN) correctly to ensure that browsers with hostname verification will
```

Comment:
I'll remove the "with hostname verification" part so that the sentence reads:
>When you obtain a server certificate, you must set its subject alternative name (SAN) correctly to ensure that browsers will trust it.